grouper/fe/handlers.py

```diff
@@ -858,6 +886,188 @@ def get(self, group_id=None, name=None):
+        form.argument_select.choices = [("", "")]
+
+        return form
+
+    def _prep_form_for_display(self, form):
+        # hack: we don't want validation to puke if one is undefined but we
+        # want them to show up in the UI as such
+        form.argument_select.flags.required = True
+        form.argument_text.flags.required = True
+
+        return form
+
+    def get(self, group_id=None, name=None):
+        group = Group.get(self.session, group_id, name)
+        if not group:
+            return self.notfound()
+
+        form = GroupPermissionRequestForm()
+        args_by_perm = get_grantable_permissions(self.session)
+        form = self._prep_form_for_display(self._prep_form(form, args_by_perm))
+
+        self.render("group-permission-request.html", form=form, group=group,
+                args_by_perm_json=json.dumps(args_by_perm))
+
+    def post(self, group_id=None, name=None):
+        group = Group.get(self.session, group_id, name)
+        if not group:
+            return self.notfound()
+
+        # only owner of group can request permissions for that group
+        role_index = self.current_user.my_role_index(group.my_members())
+        if role_index not in OWNER_ROLE_INDICES:
+            return self.forbidden()
+
+        # check inputs
+        form = GroupPermissionRequestForm(self.request.arguments)
+        args_by_perm = get_grantable_permissions(self.session)
+
+        form = self._prep_form(form, args_by_perm)
+
+        if not form.validate():
+            form = self._prep_form_for_display(form)
+            return self.render(
+                    "group-permission-request.html", form=form, group=group,
+                    args_by_perm_json=json.dumps(args_by_perm),
+                    alerts=self.get_form_alerts(form.errors),
+                    )
+
+        # load requested permission
+        if form.argument_select.data:
+            argument = form.argument_select.data
+        else:
+            argument = form.argument_text.data
+        permission = Permission.get(self.session, form.permission_name.data)
+
+        # make sure group doesn't already have this permission
+        for _, permission_name, _, permission_argument, _ in group.my_permissions():
+            if permission.name == permission_name and argument == permission_argument:
+                return self.render(
+                        "group-permission-request.html", form=form, group=group,
+                        args_by_perm_json=json.dumps(args_by_perm),
+                        alerts=[
+                            Alert("danger", "this group already has this permission+argument pair"),
+                            ],
+                        )
+
+        # make sure there's an owner for this
+        calc_args = args_by_perm.get(permission.name, [])
+        if not permission or (argument not in calc_args and "*" not in calc_args):
```

Comment:
If `permission` is `None` (so the first part of this conditional fails), won't the previous line explode?